cognite/client/data_classes/data_modeling/query.py

```diff
@@ -449,32 +449,59 @@ def get_edges(self, result_expression: str) -> EdgeListWithCursor:
         )
 
 
-# class SetOperationResultSetExpression(ResultSetExpression):
-#     ...
-#
-#
-# class UnionAllTableExpression(SetOperationResultSetExpression):
-#     def __init__(
-#         self, union_all: list[QuerySetOperationTableExpression | str], except_: list[str] = None, limit: int = None
-#     ):
-#         self.union_all = union_all
-#         self.except_ = except_
-#         self.limit = limit
-#
-#
-# class UnionTableExpression(SetOperationResultSetExpression):
-#     def __init__(
-#         self, union: list[QuerySetOperationTableExpression | str], except_: list[str] = None, limit: int = None
-#     ):
-#         self.union = union
-#         self.except_ = except_
-#         self.limit = limit
-#
-#
-# class IntersectTableExpression(SetOperationResultSetExpression):
-#     def __init__(
-#         self, intersection: list[QuerySetOperationTableExpression | str], except_: list[str] = None, limit: int = None
-#     ):
-#         self.intersect = intersection
-#         self.except_ = except_
-#         self.limit = limit
+class SetOperation(ResultSetExpression, ABC):
```

Comment:
For these classes, you need a unit test for a nested `SetOperation` and a custom dump method as the default dump method from `CogniteObject` does consider nested objects.

Note that the automatic testing of dump/load does not account for recursion.

```python
assert SetOpertation.load({"union": [{"intersection": ["0", "1"]}, "4"] }).dump() == {"union": [{"intersection": ["0", "1"]}, "4"] }
```